File: solutions/Leetcode_848/leetcode_848.cpp
#include <string>
#include <vector>

class Solution1
{
public:
    std::string shiftingLetters(std::string s, std::vector<int> &shifts)
    {
        int prevShift = 0;
        std::string shiftedLetters = s;
        for (int i = shifts.size() - 1; i >= 0; i--)
        {
            int totalShift = (prevShift + shifts[i]) % 26;
            prevShift = totalShift;
            shiftedLetters[i] = (shiftedLetters[i] - 'a' + totalShift) % 26 + 'a';
        }
        return shiftedLetters;
    }
};

File: solutions/Leetcode_848/leetcode_848.py
from typing import List


class Solution1:
    def shifting_letters(self, s: str, shifts: List[int]) -> str:
        prev_shift = 0
        ret = []
        for i in range(len(s)-1, -1, -1):
            total_shift = prev_shift + shifts[i]
            prev_shift = total_shift
            ret.append(chr((ord(s[i]) + total_shift - 97) % 26 + 97))
        return ''.join(ret[::-1])
